src/Avalonia.X11/X11TrayIconImpl.cs

```diff
@@ -0,0 +1,369 @@
+
+            for (var i = 0; i < pixLength; i++)
+            {
+                var u = rx[i].ToUInt32();
+                pixByteArray[pixByteArrayCounter++] = (byte)((u & 0xFF000000) >> 24);
+                pixByteArray[pixByteArrayCounter++] = (byte)((u & 0xFF0000) >> 16);
+                pixByteArray[pixByteArrayCounter++] = (byte)((u & 0xFF00) >> 8);
+                pixByteArray[pixByteArrayCounter++] = (byte)(u & 0xFF);
+            }
+
+            _icon = new DbusPixmap(w, h, pixByteArray);
+            _statusNotifierItemDbusObj.SetIcon(_icon);
+        }
+
+        public void SetIsVisible(bool visible)
+        {
+            if (_con is null || _isDisposed || !_ctorFinished) return;
+
+            if (visible & !_isActive)
+            {
+                DestroyTrayIcon();
+                CreateTrayIcon();
+            }
+            else if (!visible & _isActive)
+            {
+                DestroyTrayIcon();
+            }
+        }
+
+        public void SetToolTipText(string text)
+        {
+            if (_con is null || _isDisposed) return;
+            _tooltipText = text;
+            _statusNotifierItemDbusObj?.SetTitleAndTooltip(_tooltipText);
+        }
+    }
+
+    /// <summary>
+    /// DBus Object used for setting system tray icons.
+    /// </summary>
+    /// <remarks>
+    /// Useful guide: https://web.archive.org/web/20210818173850/https://www.notmart.org/misc/statusnotifieritem/statusnotifieritem.html
+    /// </remarks>
+    internal class StatusNotifierItemDbusObj : IStatusNotifierItem
+    {
+        private readonly StatusNotifierItemProperties _backingProperties;
+        private event Action<PropertyChanges> OnPropertyChange;
+        public event Action OnTitleChanged;
+        public event Action OnIconChanged;
+        public event Action OnAttentionIconChanged;
+        public event Action OnOverlayIconChanged;
+        public event Action OnTooltipChanged;
+        public Action<INativeMenuExporter> SetNativeMenuExporter { get; set; }
+        public Action<string> NewStatusAsync { get; set; }
+        public Action ActivationDelegate { get; set; }
+        public ObjectPath ObjectPath { get; }
+
+        public StatusNotifierItemDbusObj(ObjectPath dbusmenuPath)
+        {
+            ObjectPath = new ObjectPath($"/StatusNotifierItem");
+
+            _backingProperties = new StatusNotifierItemProperties
+            {
+                Menu = dbusmenuPath, // Needs a dbus menu somehow
+                ToolTip = new ToolTip("")
+            };
+
+            InvalidateAll();
+        }
+
+        public Task ContextMenuAsync(int X, int Y) => Task.CompletedTask;
+
+        public Task ActivateAsync(int X, int Y)
+        {
+            ActivationDelegate?.Invoke();
+            return Task.CompletedTask;
+        }
+
+        public Task SecondaryActivateAsync(int X, int Y) => Task.CompletedTask;
+
+        public Task ScrollAsync(int Delta, string Orientation) => Task.CompletedTask;
```

Comment:
Method parameters names should be in camelCase.Or maybe it is intentional and need by DBus mapping? If so, maybe add comments so we wouldn't break it.

Reply:
no it's not that sensitive on the param names i think, we can make it camel case